File: objc/2-variables.m
#import <Foundation/Foundation.h>
// comment

/* 
 * long comment
 * on multiple lines
 */

int main()
{
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];

    char * string ="hello";
    NSString * string1 = @"string1";
    int number=123;
    char character='a';
    int * pointer=&number;
    enum month {
        january,
        february,
        may=5,
        june,
    };

    NSLog(@"string=%s string1=%@ number=%d character=%c pointer=%p value at pointer=%d\n", 
            string, string1, number, character, pointer, *pointer);
    *pointer = 456;
    NSLog(@"string=%s number=%d character=%c pointer=%p value at pointer=%d\n", string, number, character, pointer, *pointer);

    enum month s=january;
    enum month m = may;
    enum month c = june;

    NSLog(@"months:  january=%d may=%d june=%d\n",s, m, c);
    [pool drain];
    return 0;
}

File: rust/2-variables.rs
fn main() {
    let string = "hello";
    let character = 'a';

    let number1 = 123;
    let mut number2 = 124;

    let pointer1=&number1 as * const i32;
    let pointer2=&mut number2 as * mut i32;

    unsafe {
        println!("string={} number1={} number2={} character={}", string, number1, number2, character);
        println!("pointer1={:p} value at pointer1={}",pointer1, *pointer1);
        println!("pointer2={:p} value at pointer2={}",pointer2, *pointer2);

        *pointer2 = 456;
        println!("pointer2={:p} value at pointer2={}",pointer2, *pointer2);
    }

    enum Month {
        January,
//        February,
        May = 5,
        June,
    }

    let s = Month::January;
    let m = Month::May;
    let c = Month::June;

    println!("months: January={:?} May={:?} June={:?}", s as i32, m as i32, c as i32);
}
